source/common/protobuf/visitor.cc

```diff
@@ -1,25 +1,45 @@
 #include "source/common/protobuf/visitor.h"
 
+#include "source/common/protobuf/utility.h"
+
 namespace Envoy {
 namespace ProtobufMessage {
 
 void traverseMessage(ConstProtoVisitor& visitor, const Protobuf::Message& message,
-                     const void* ctxt) {
-  visitor.onMessage(message, ctxt);
+                     bool recurse_into_any) {
+  visitor.onMessage(message);
+
+  // If told to recurse into Any messages, do that here and skip the rest of the function.
+  if (recurse_into_any && message.GetDescriptor()->full_name() == "google.protobuf.Any") {
```

Comment:
What about when we used TypedStruct? I think one of the reasons we might have done validation bottom-up previously was because we used Struct, where we didn't have reflection based typing. Now that we no longer support Struct, things are better, but we do have TypedStruct as an alternative to Any we would need to handle.

Reply:
Thanks for the reminder. I will add this support.